### PR TITLE
app-misc/lirc-0.9.4c: version bump

### PR DESCRIPTION
**Pull request by candrews**
Also Use xdg-utils.eclass to cleanup XDG_* env causing sandbox violation

https://bugs.gentoo.org/show_bug.cgi?id=602978
https://bugs.gentoo.org/show_bug.cgi?id=587770

### CODE REVIEW

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+PATCHES=(
+	"${FILESDIR}"/lirc-0.9.4-doc-path-fix.patch # https://bugs.gentoo.org/show_bug.cgi?id=589246 https://sourceforge.net/p/lirc/tickets/214/
+)
+
+src_configure() {
+	xdg_environment_reset
+	filter-flags -Wl,-O1
+	econf \
+		--localstatedir=/var \
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: missing `${EPREFIX}`

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	newconfd "${FILESDIR}"/lircmd.conf lircmd
+
+	insinto /etc/modprobe.d/
+	newins "${FILESDIR}"/modprobed.lirc lirc.conf
+
+	newinitd "${FILESDIR}"/irexec-initd-0.9.4a-r2 irexec
+	newconfd "${FILESDIR}"/irexec-confd irexec
+
+	keepdir /etc/lirc
+	if [[ -e "${D}"/etc/lirc/lircd.conf ]]; then
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: in general, avoid `${D}` for "free-form stuff", i.e. with rm etc. Use `${ED}` instead, and watch out for double-slashing -> Use `"${ED%/}"/`

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		newdoc "${D}"/etc/lirc/lircd.conf lircd.conf.example
+	fi
+
+	find "${D}" -name '*.la' -delete || die
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: you can leave `"${D}"` here

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	# Avoid QA notice
+	rm -d "${D}"/var/run/lirc || die
+	rm -d "${D}"/var/run || die
+}
+
+pkg_preinst() {
+	local dir="${EROOT}/etc/modprobe.d"
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: `EROOT` also needs to be in the form of `"${EROOT%/}"/`, see also the last section of https://devmanual.gentoo.org/ebuild-writing/variables/

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	if [[ -e "${T}"/lircd.conf ]]; then
+		cp "${T}"/lircd.conf "${EROOT}"/etc/lirc/lircd.conf || die
+		if [[ "$MOVE_OLD_LIRCD_CONF" = "1" ]]; then
+			elog "Moved /etc/lircd.conf to /etc/lirc/lircd.conf"
+			rm -f "${EROOT}"/etc/lircd.conf || die
+		fi
+	fi
+
+	einfo "The new default location for lircd.conf is inside of"
+	einfo "/etc/lirc/ directory"
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: Add `EROOT` to message

**Comment by KungFuJesus** [comment]
Might this fix the issue it has recently building with more recent kernel headers?
https://bugs.gentoo.org/show_bug.cgi?id=599680
If not, I suggest perhaps adding a patch for that, as well into this ebuild.

**Comment by candrews** [comment]
0.9.4c does indeed fix that issue

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 5 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: Thanks!

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 4 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: I bet `FILESDIR` should have the same treatment, so I made the same change for it.

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 2 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: makes sense, thank you. Done.

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 1 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: done

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	)
+	irman? ( media-libs/libirman )
+	iguanair? ( app-misc/iguanaIR )
+	ftdi? ( dev-embedded/libftdi:0 )
+	inputlirc? ( app-misc/inputlircd )
+	usb? ( virtual/libusb:0 )
+"
+
+PATCHES=(
+	"${FILESDIR}"/lirc-0.9.4-doc-path-fix.patch # https://bugs.gentoo.org/show_bug.cgi?id=589246 https://sourceforge.net/p/lirc/tickets/214/
```
Comment (Polynomial-C) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: Please move comments above the actual patch and put each URL in a separate line.

Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: 👍 done

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+"
+
+PATCHES=(
+	# https://bugs.gentoo.org/show_bug.cgi?id=589246 https://sourceforge.net/p/lirc/tickets/214/
+	"${FILESDIR}"/lirc-0.9.4-doc-path-fix.patch
+)
+
+src_configure() {
+	xdg_environment_reset
+	filter-flags -Wl,-O1
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: out of curiosity, why is this needed? is there some underlinking somewhere?

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 2 above)
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: Are you still going to change these `${D}`'s?

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+		mv -f "${dir}/lirc" "${dir}/lirc.conf" || die
+	fi
+
+	# copy the first file that can be found
+	if [[ -f "${EROOT%/}"/etc/lirc/lircd.conf ]]; then
+		cp "${EROOT%/}"/etc/lirc/lircd.conf "${T}"/lircd.conf || die
+	elif [[ -f "${EROOT%/}"/etc/lircd.conf ]]; then
+		cp "${EROOT%/}"/etc/lircd.conf "${T}"/lircd.conf || die
+		MOVE_OLD_LIRCD_CONF=1
+	elif [[ -f "${D}"/etc/lirc/lircd.conf ]]; then
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: `ED`

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 7 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: That was there when I started working on this ebuild, so I didn't question it. I removed it and the there's no apparent negative consequence. Upstream has done a bit of work on their build system lately, so I'm assuming filtering these flags was necessary for an earlier lirc version but no longer is.

File: app-misc/lirc/lirc-0.9.4c.ebuild (hunk 3 above)
Comment (candrews) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: For consistency, would it better to use ED instead of having this be the only place D is used?

File: app-misc/lirc/lirc-0.9.4c.ebuild [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+KEYWORDS="~amd64 ~ppc ~ppc64 ~x86"
+IUSE="doc static-libs X audio irman ftdi inputlirc iguanair systemd usb"
+
+S="${WORKDIR}/${MY_P}"
+
+REQUIRED_USE="
+	iguanair? ( irman )
+"
+DEPEND="
+	doc? ( app-doc/doxygen )
```
Comment (SoapGentoo) [inline app-misc/lirc/lirc-0.9.4c.ebuild]: all the deps in `RDEPEND` aren't required at compile time? Really?
